src/main/java/rx/internal/util/IndexedRingBuffer.java

```diff
@@ -47,6 +47,10 @@
  * @param <E>
  */
 public final class IndexedRingBuffer<E> implements Subscription {
+    private final ElementSection<E> elements = new ElementSection<E>();
+    private final IndexSection removed = new IndexSection();
+    /* package for unit testing */final AtomicInteger index = new AtomicInteger();
```

Comment:
Maybe worthwhile to import IntelliJ annotations or create a project annotation `@VisibleForTesting`?